W-14146679 dw troubleshoot duke (#210)
mulesoft/docs-code-builder: modules/ROOT/pages/troubleshoot-dataweave-errors.adoc

= DataWeave Errors
:page-deployment-options: cloud-ide, desktop-ide

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//LOGO (web, desktop, or both)
// include::partial$acb-ide-logos.adoc[tags="both-ides"]

In Anypoint Code Builder, DataWeave errors can appear in the *Problems* area of the output panel and within a configuration XML file.

> == Hundreds or Thousands of DataWeave Problems in the Output Panel

Author: ```suggestion
== Differences between DataWeave Problems in Studio and Anypoint Code Builder in the Output Panel
```

> == Hundreds or Thousands of DataWeave Problems in the Output Console

Author: ```suggestion
== Hundreds or Thousands of DataWeave Problems in the Output Panel
```

> A list of hundreds or thousands of DataWeave issues sometimes appears in the *Problems* area of the output panel after you import a project from Anypoint Studio (Studio) into the Anypoint Code Builder cloud or desktop IDE. 

Author: ```suggestion
When you import a project from Anypoint Studio to Anypoint Code Builder, a list of hundreds or thousands of DataWeave issues sometimes appears in the *Problems* area of the output panel after you import a project from Anypoint Studio (Studio) into the Anypoint Code Builder cloud or desktop IDE. 
```

> Typically, these issues are warnings related to metadata propagation that Studio hides but aggregates into a single error message, `Failure to provide Metadata propagation`. Unlike Studio, Anypoint Code Builder displays the hidden warnings as problems because the issues appear as script warnings in the configuration XML for the project. The problem message often start with the text `Unable to resolve the reference of`.

Author: ```suggestion
Typically, these issues are warnings related to metadata propagation that Studio hides but aggregates into a single error message, `Failure to provide Metadata propagation`. Unlike Studio, Anypoint Code Builder displays the hidden warnings as problems because the issues appear as script warnings in the configuration XML for the project. The problem message often begins with the text `Unable to resolve the reference of`.
```
